File: src/main/kotlin/meteor/plugins/xptracker/worldmapwalker/WorldMapWalkerConfig.java
package meteor.plugins.xptracker.worldmapwalker;

import meteor.config.Config;
import meteor.config.ConfigGroup;
import meteor.config.ConfigItem;

@ConfigGroup("worldmap")
public interface WorldMapWalkerConfig extends Config {

    @ConfigItem(
            keyName = "shiftClickToWalk",
            name = "Shift click on map to walk",
            description = "Lets you walk to destinations with shift clicks"
    )
    default boolean shiftClickToWalk() {
        return true;
    }

    @ConfigItem(
            keyName = "showPaths",
            name = "Show walk path",
            description = "Highlights the walking path in game"
    )
    default boolean showPaths() {
        return true;
    }
}

File: client/src/main/kotlin/meteor/plugins/worldmapwalker/WorldMapWalkerConfig.kt
package meteor.plugins.worldmapwalker


import meteor.config.legacy.Config
import meteor.config.legacy.ConfigGroup
import meteor.config.legacy.ConfigItem

@ConfigGroup("worldmap")
interface WorldMapWalkerConfig : Config {
    @ConfigItem(
        keyName = "shiftClickToWalk",
        name = "Shift click on map to walk",
        description = "Lets you walk to destinations with shift clicks"
    )
    fun shiftClickToWalk(): Boolean {
        return true
    }

    @ConfigItem(keyName = "showPaths", name = "Show walk path", description = "Highlights the walking path in game")
    fun showPaths(): Boolean {
        return true
    }
}